common/types.go

```diff
@@ -173,17 +174,20 @@ type LogConfig struct {
 }
 
 type Block struct {
+	sync.RWMutex
 	StartIndex int64
 	EndIndex   int64
-	Size       int64
+	Data       []byte
+	Dirty      bool
```

Comment:
In common we have a bitmap we can use it here instead of maintaining multiple flags. That will conserve memory and also addition of new flags in future will not require additional variables in struct. See HandleMap implementation, it was converted to same model.